### ISSUE TITLE
Regression: out geom(bbox) doesn't return geometry

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#417]
Issue: If a way is not fully included in a bounding box, out geom(bbox) doesn't return any geometry anymore. This affects the query tool on osm.org: https://github.com/openstreetmap/openstreetmap-website/issues/1608

Test case:

```
[timeout:10][out:json];way(193407337);
out geom(43.44416399866694,-6.424341201782227,43.458401741026194,-6.399600505828857);
```

## Previous behavior:

Partial geometry returned:

![grafik](https://user-images.githubusercontent.com/5842757/29192908-872ecd38-7e23-11e7-8cd1-f2e9ac2e6085.png)

## Current behavior on overpass-api.de:

Only bounding box returned, no geometry at all:

![grafik](https://user-images.githubusercontent.com/5842757/29192932-9a97a19c-7e23-11e7-991a-55db963b39ef.png)

**Comment by tyrasd** [comment]
The regression seems to be only affecting the `out:json` mode, while it still works as expected in xml output mode: http://overpass-turbo.eu/s/qVR

**Comment by mmd-osm** [comment]
Good observation. It's probably this line here: https://github.com/drolbr/Overpass-API/blob/master/src/overpass_api/output_formats/output_json.cc#L168 - Partial way geometries return false for `has_line_geometry()` (see: https://github.com/drolbr/Overpass-API/blob/350cf560956c669171b949b162f239b55936e1a8/src/overpass_api/core/geometry.h#L262), that probably needs to be changed to has_faithful_way_geometry (Bonus points for finding out what "faithful" really means in this context, it's not documented).

**Comment by mmd-osm** [comment]
Fix looks ok here: http://overpass-turbo.eu/s/qXb

**Comment by drolbr** [comment]
Fixed in 2f085a7e522f187353f81734bcb2ba9991e99fb6.
